src/molecule.cpp

```diff
@@ -375,15 +369,17 @@ void MoleculeStructureReader::readArray(ParticleVector &particles, const json &j
     }
 }
 
-void MoleculeStructureReader::readFasta(ParticleVector &particles, const json &j_fasta) {
-    if (j_fasta.find("fasta") == j_fasta.end()) {
+void MoleculeStructureReader::readFasta(ParticleVector &particles, const json &input) {
+    if (auto it = input.find("fasta"); it == input.end()) {
         throw ConfigurationError("invalid FASTA format");
+    } else {
```

Comment:
As `it` is accessible in the whole method anyway, the if-init statement oddly introducing the else block is not really needed.

```
auto it = …;
if (it == input.end()) {
    throw …;
}
// useful code
```

Reply:
Thanks for the review! I'm trying to develop a habit of reducing variable scope using C++17's if-statement with an initialiser. But I agree it doesn't add a lot of safety here.